pkg/util/instancemetadata/prod_test.go

```diff
@@ -205,7 +205,7 @@ func TestPopulateTenantIDFromMSI(t *testing.T) {
 					OAuthToken().
 					Return("invalid")
 			},
-			wantErr: "token contains an invalid number of segments",
+			wantErr: "400: InvalidServicePrincipalToken: properties.servicePrincipalProfile: token contains an invalid number of segments",
```

Comment:
Can you please replace in the error message `token contains an invalid number of segments` with something that's more meaningful to end customers in this context, like `The provided service principal is invalid` ?

Reply:
@rogbas Yes of course, I'll update the PR.